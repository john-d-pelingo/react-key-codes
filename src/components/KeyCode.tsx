import { css } from 'emotion'
import React from 'react'

interface IKeyCodeProps {
  handleClick?: (event: React.MouseEvent<HTMLSpanElement>) => void
  keyCode: number
  keyText?: string
}

export const KeyCode: React.FC<IKeyCodeProps> = ({
  keyCode,
  keyText,
  handleClick = () => undefined,
}) => (
  <div className={keyCodeCss}>
    <span aria-label="key-code-number" className="key-code-number">
      {keyCode}
    </span>
    <button
      aria-label="key-code-button"
      className="key-code-button"
      onClick={handleClick}
    >
      {keyText || 'What key code is that?'}
    </button>
  </div>
)

const keyCodeCss = css`
  display: table-cell;
  vertical-align: middle;

  .key-code-number {
    display: block;
    font-size: 300px;
    line-height: 320px;
    padding-bottom: 40px;
  }

  .key-code-button {
    background: #eff0f2;
    border-radius: 4px;
    border-top: 1px solid rgba(255, 255, 255, 0.8);
    box-shadow: inset 0 0 25px #e8e8e8, 0 1px 0 #c3c3c3, 0 2px 0 #c9c9c9,
      0 2px 3px #333;
    color: #aaa;
    cursor: pointer;
    display: inline-block;
    font: 38px/40px 'Helvetica Neue';
    font-weight: bold;
    margin: 0;
    padding: 15px 30px;
    position: relative;
    text-align: center;
    text-decoration: none;
    text-shadow: 0 1px 0 rgba(255, 255, 255, 0.8);
  }
`
