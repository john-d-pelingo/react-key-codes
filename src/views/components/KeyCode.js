import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    keyCode: PropTypes.number,
    keyText: PropTypes.string.isRequired
};

const defaultProps = {
    keyCode: null
};

const KeyCode = ({ keyCode, keyText }) => (
    <div className="key-code">
        <span className="key-code-code">{ keyCode }</span>
        <span className="key-code-text">{ keyText || 'What key code is that?' }</span>
    </div>
);

KeyCode.propTypes = propTypes;
KeyCode.defaultProps = defaultProps;

export default KeyCode;
