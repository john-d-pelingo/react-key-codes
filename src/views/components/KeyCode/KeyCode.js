/* eslint-disable jsx-a11y/no-static-element-interactions */

import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  keyCode: PropTypes.number,
  keyText: PropTypes.string.isRequired,

  handleClick: PropTypes.func.isRequired
};

const defaultProps = {
  keyCode: null
};

const KeyCode = ({ keyCode, keyText, handleClick }) => (
  <div className="key-code">
    <span className="key-code-code">{ keyCode }</span>
    <span className="key-code-text" onClick={ handleClick }>{ keyText || 'What key code is that?' }</span>
  </div>
);

KeyCode.propTypes = propTypes;
KeyCode.defaultProps = defaultProps;

export default KeyCode;
