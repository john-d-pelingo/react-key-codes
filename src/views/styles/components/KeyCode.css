.key-code {
    display        : table-cell;
    vertical-align : middle;
}

.key-code .key-code-code {
    display        : block;
    font           : 150px/160px 'Raleway', 'Helvetica Neue', 'Helvetica', Arial, sans-serif;
    padding-bottom : 40px;
}

.key-code .key-code-text {
    background      : #eff0f2;
    border-radius   : 4px;
    border-top      : 1px solid rgba(255, 255, 255, 0.8);
    box-shadow      : inset 0 0 25px #e8e8e8, 0 1px 0 #c3c3c3, 0 2px 0 #c9c9c9, 0 2px 3px #333;
    color           : #aaa;
    cursor          : pointer;
    display         : inline-block;
    font            : 38px/40px 'Helvetica Neue', 'Helvetica', Arial, sans-serif;
    font-weight     : bold;
    margin          : 0;
    padding         : 15px 30px;
    position        : relative;
    text-align      : center;
    text-decoration : none;
    text-shadow     : 0 1px 0 rgba(255, 255, 255, 0.8);
}
