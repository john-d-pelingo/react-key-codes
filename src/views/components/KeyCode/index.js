import KeyCode from './KeyCode';

export default KeyCode;
