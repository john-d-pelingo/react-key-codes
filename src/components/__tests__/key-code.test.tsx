import React from 'react'

import KeyCode from '../key-code'

describe.skip(`${KeyCode.name} Component`, () => {
  it('should with correct styles', () => {
    const defaultProps = {
      keyCode: null,
      keyText: '',
      handleClick: () => {},
    }

    const snap = mount(<KeyCode {...defaultProps} />)
    const snapJson = enzymeToJson(snap)

    expect(snapJson).toMatchSnapshot()
  })
})
