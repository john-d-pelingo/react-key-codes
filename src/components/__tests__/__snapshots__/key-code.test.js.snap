// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyCode Component should with correct styles 1`] = `
.emotion-0 {
  display: table-cell;
  vertical-align: middle;
}

.emotion-0 .key-code-code {
  display: block;
  font-size: 300px;
  line-height: 320px;
  padding-bottom: 40px;
}

.emotion-0 .key-code-text {
  background: #eff0f2;
  border-radius: 4px;
  border-top: 1px solid rgba(255,255,255,0.8);
  box-shadow: inset 0 0 25px #e8e8e8,0 1px 0 #c3c3c3,0 2px 0 #c9c9c9, 0 2px 3px #333;
  color: #aaa;
  cursor: pointer;
  display: inline-block;
  font: 38px/40px 'Helvetica Neue';
  font-weight: bold;
  margin: 0;
  padding: 15px 30px;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: 0 1px 0 rgba(255,255,255,0.8);
}

<KeyCode
  handleClick={[Function]}
  keyCode={null}
  keyText=""
>
  <div
    className="emotion-0"
  >
    <span
      className="key-code-code"
    />
    <span
      className="key-code-text"
      onClick={[Function]}
    >
      What key code is that?
    </span>
  </div>
</KeyCode>
`;
